NextcloudTalk/BaseChatViewController.swift

```diff
@@ -475,14 +475,41 @@ import QuickLook
         temporaryMessage.actorType = "users"
         temporaryMessage.timestamp = Int(Date().timeIntervalSince1970)
         temporaryMessage.token = room.token
-        temporaryMessage.message = self.replaceMentionsDisplayNamesWithMentionsKeysInMessage(message: message, parameters: messageParameters)
 
         let referenceId = "temp-\(Date().timeIntervalSince1970 * 1000)"
         temporaryMessage.referenceId = NCUtils.sha1(fromString: referenceId)
         temporaryMessage.internalId = referenceId
         temporaryMessage.isTemporary = true
+        temporaryMessage.isOfflineMessage = true
         temporaryMessage.parentId = parentMessage?.internalId
-        temporaryMessage.messageParametersJSONString = messageParameters
+
+        if isVoiceMessage {
+            var messageParametersDict = [String: Any]()
+            let parameterId = UUID().uuidString
+
+            temporaryMessage.message = message
+            temporaryMessage.messageType = kMessageTypeVoiceMessage
+
+            let fileParameterDict: [String: Any] = [
+                "id": parameterId,
+                "type": "file",
+                "name": message,
+                "path": messageParameters,
+                "fileId": parameterId,
+                "fileName": message,
+                "filePath": messageParameters,
+                "fileLocalPath": messageParameters
+            ]
+
+            messageParametersDict[parameterId] = fileParameterDict
```

Comment:
Why using a UUID here? Should it be the file dictionary? I don't see it used anywhere? (Maybe I missed it, not yet through everything).

Reply:
Good question, I guess I had a problem with the file parameter not being set when putting the fileParameters in no dictionary.
What's your suggestion?